uwh-refbox/src/main.rs

```diff
@@ -140,53 +142,201 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
+                    GamePeriod::FirstHalf | GamePeriod::OvertimeFirstHalf => {
+                        ("1ST", 72, 33, "HALF", 68, 48)
+                    }
+                    GamePeriod::SecondHalf | GamePeriod::OvertimeSecondHalf => {
+                        ("2ND", 72, 33, "HALF", 68, 48)
+                    }
+                    _ => ("PERIOD ERROR", 72, 33, "", 0, 0),
+                };
+
+                display.draw(
+                    Font8x15::render_str(text1)
+                        .stroke(Some(game_color))
+                        .translate(Point::new(x1, y1)),
+                );
+                display.draw(
+                    Font8x15::render_str(text2)
+                        .stroke(Some(game_color))
+                        .translate(Point::new(x2, y2)),
+                );
+
+                let (text1, x1, y1, text2, x2, y2, text3, x3, y3) = match state.timeout {
+                    TimeoutState::White(secs) => (
+                        "WHITE",
+                        76,
+                        2,
+                        "TIMEOUT",
+                        68,
+                        17,
+                        format!(":{}", secs),
+                        132,
+                        2,
+                    ),
+                    TimeoutState::Black(secs) => (
+                        "BLACK",
+                        76,
+                        2,
+                        "TIMEOUT",
+                        68,
+                        17,
+                        format!(":{}", secs),
+                        132,
+                        2,
+                    ),
+                    TimeoutState::Ref(_) => ("REF TIMEOUT", 68, 4, "", 0, 0, String::new(), 0, 0),
+                    TimeoutState::PenaltyShot(_) => {
+                        ("PENALTY", 65, 4, "SHOT", 149, 4, String::new(), 0, 0)
+                    }
+                    _ => ("T/O ERROR", 76, 2, "", 0, 0, String::new(), 0, 0),
+                };
+                display.draw(
+                    Font8x15::render_str(text1)
+                        .stroke(Some(timeout_color))
+                        .translate(Point::new(x1, y1)),
+                );
+                display.draw(
+                    Font8x15::render_str(text2)
+                        .stroke(Some(timeout_color))
+                        .translate(Point::new(x2, y2)),
+                );
+                display.draw(
+                    Font16x31::render_str(&text3)
+                        .stroke(Some(timeout_color))
+                        .translate(Point::new(x3, y3)),
+                );
+                display.draw(
+                    Font16x31::render_str(&secs_to_time_string(state.secs_in_period))
+                        .stroke(Some(game_color))
+                        .translate(Point::new(108, 33)),
+                );
+            }
+        };
+
+        let black_penalty_count = 1;
+        let white_penalty_count = 1;
```

Comment:
Should probably mark these with a comment as being interim

Reply:
ok